util/vendor.py

```diff
@@ -340,79 +352,94 @@ def main(argv):
         raise SystemExit(sys.exc_info()[1])
     desc['_base_dir'] = vendor_file_base_dir
 
+
+    desc_file_stem = desc_file_path.name.rsplit('.', 2)[0]
+    lock_file_path = desc_file_path.with_name(desc_file_stem + '.lock.hjson')
+
+    # Importing may use lock file upstream, information, so make a copy now
+    # which we can overwrite with the upstream information from the lock file.
+    import_desc = desc.copy()
+
     # Load lock file contents (if possible)
-    desc_file_path_str = str(desc_file_path)
-    lock_file_path = Path(
-        desc_file_path_str[:desc_file_path_str.find('.vendor.hjson')] +
-        '.lock.hjson')
-    try:
-        with open(lock_file_path, 'r') as f:
+    if lock_file_path.is_file():
```

Comment:
The try/catch approach is more common in Python and race-condition/corner-case-safer ("it's easier to ask for forgiveness than permission"), but fine either way.

Reply:
Addressed.